resources/js/features/root/pages/Calendar.page.tsx

```diff
@@ -34,19 +34,19 @@ export default (() => {
       <SegmentedControl
         name="calModeControl"
         items={[
-          {text: "Miesiąc", value: "month"},
-          {text: "Tydzień", value: "week"},
-          {text: "Dzień", value: "day"},
+          {value: "month", label: () => <>Miesiąc</>},
```

Comment:
Ubrzydziło się trochę. Nie chcesz jakiegoś ```label: string | JSXElement```?

Dlaczego w ogóle () => JSXElement? JSXElement sam w sobie już jest reaktywny, to by nie wystarczyło? (np. komponenty go zwracają)

Reply:
Moim zdaniem nie tak bardzo ubrzydziło, zresztą to jest śmieciowy kod, i w ogóle to być może string jest JSX sam z siebie, tyle że niereaktywny.

Jeśli chodzi o tę funkcję to to jest efekt użerania się już jakiś czas temu z którymś innym komponentem, chyba headerem tabeli. W szczególności element JSX jest jak element, i może być tylko w jednym miejscu. I się czasem okazywało że nagle potrzebuję dwóch kopii, a nie mogę mieć bo się coś rypie. I tak mnie to zmęczyło że ustaliłem żeby zawsze dawać `()=>JSX.Element`, który to typ zresztą nazywa się Component, więc to nie jest od rzeczy, tyle że nie chcę tego tak nazywać bo to wygląda jakby tu należało podać nazwę komponentu. A można, ale nie trzeba, i to zbędne zaciemnianie. Więc moim zdaniem tak jest dobrze, opóźnia tworzenie tego elementu do odpowiedniego czasu gdy jest faktycznie potrzebny, i jakby była potrzeba to działa wielokrotnie.